### ISSUE TITLE
Misfire `publishAdd` if vary associations between two models

### ISSUE THREAD

**Issue by clarkorz** [balderdashy/sails#1727]
Try these models:

    // api/models/User.js
    module.exports = {
      attributes: {
        name: 'STRING',
        age : 'INTEGER',
        pets: { collection: 'pet', via: 'owners' },
        feed: { model: 'Pet' }
      }
    };

    // api/models/Pet.js
    module.exports = {
      attributes: {
        name:'STRING',
        color:'STRING',
        master: { model: 'User' },
        feedBy: { model: 'User' },
        owners: { collection: 'User', via: 'pets', dominant: true }
      }
    };

`Pet` model has three association with `User`: an *One-Way* **One-to-One**, an *Two-Way* **One-to-One**, and a **Many-to-Many**.
    
    // assume there is a user with id=1, and create a Pet without owners but a master
    Pet.create({name:'peny', master: 1})

This will fire a `publishAdd` (`/Developer/Projects/WebStormProjects/DemoProjects/SailsProjects/sails-rc7/node_modules/sails/lib/hooks/pubsub/index.js` line 1178). `master` and `feedBy` will both treat `pets` as their *reserve association*, but actually not, and `master` even doesn't have a *reserve association* in `User` model.

When I try to fix this, I find a few problems:

1. `via` is optional [FAQ](https://github.com/balderdashy/sails-docs/blob/master/reference/ModelAssociations.md#faq), in this case, there is no way to figure out if there **is** a reserve association or which one is (`pets` or `feed`).

2. line 1175-1176, there is a comment:
> If this is a many-to-many association, do a publishAdd for the other side.

But, first, the code shows that only **One-to-Many** and **One-to-One** associations will fire a publishAdd (see line 1164). I don't know which one is intended to, the code or the comment.

**Comment by mikermcneil** [comment]
Thanks for posting, @clarkorz.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
